pkg/ride/generate/internal/constructors_generation.go

```diff
@@ -0,0 +1,290 @@
+package internal
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/wavesplatform/gowaves/pkg/ride/ast"
+)
+
+func constructorName(act actionsObject) string {
+	return strings.ToLower(string(act.StructName[0])) + act.StructName[1:] + "Constructor"
+}
+
+func argVarName(act actionField) string {
+	return act.Name
+}
+
+func extractConstructorArguments(args []actionField) ([]actionField, error) {
+	arguments := []actionField{}
+	seenOrders := map[int]bool{}
```

Comment:
May be replace type of `seenOrders` with `map[int]struct{}` and check key existence in the map. Looks like bool value can't be `false`.

Reply:
> Looks like bool value can't be false.

Yes, it is. But as for me, it seems to be more convinient way for defining a set. i mean `found := seenOrders[1]` looks better than `_, found := seenOrders[1]`, of course if we don't worry about memory usage optimization. In this case we don't because it's just a utility for code generation.

Reply:
In fact, `map[int]struct{}` is more efficient than `map[int]bool{}`. Try to compare `unsafe.Sizeof(true)` and `unsafe.Sizeof(struct{})` and you'll see the difference. And using maps with empty struct as value it's clear way to show others that this map actually is a set.

Reply:
moved to `map[int]struct{}`